src/utils.ts

```diff
@@ -545,9 +545,7 @@ export interface ClientMetadataOptions {
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const NODE_DRIVER_VERSION = require('../package.json').version;
 
-export function makeClientMetadata(options?: ClientMetadataOptions): ClientMetadata {
-  options = options ?? {};
-
+export function makeClientMetadata(options: MongoOptions): ClientMetadata {
```

Comment:
```ts
    const name = Buffer.byteLength(options.appName, 'utf8') <= 128 ? options.appName : Buffer.from(options.appName, 'utf8').subarray(0, 128).toString('utf8')
    metadata.application = { name };
```
From the other PR, can we make the improvement:

I suggest some extra reformatting here, but main points, let's use `Buffer.byteLength()` instead of converting everytime, and use subarray instead of slice since slice is deprecated.

Reply:
I'll make the refactor but to clarify - any "inefficiency" here isn't important.  This function is only invoked once per mongo client now

Reply:
I still think it has value even though it is done once, thanks!